### ISSUE TITLE
ERROR: Too many open files

### ISSUE THREAD

**Issue by kevinjqiu** [vsespb/mt-aws-glacier#56]
On master:HEAD:

A few thousand files in my backup folder to sync.  mtglacier ran for about an hour and failed with:

```
ERROR (parent): Unable to open task file "/path/to/file" for reading, errno=Too many open files
```

Fails under both single and multiprocess.

The operating system I'm running it on is the latest Raspbian on Raspberry Pi.

EDIT: Add more environment info.

**Comment by vsespb** [comment]
Hi. What version?

**Comment by kevinjqiu** [comment]
> Hi. What version?

master:HEAD

Also, the host operating system is Debian on Raspberry Pi.  I've updated the issue description.  Thanks!

**Comment by vsespb** [comment]
ok, I think I reproduced problem, seems new bug (resource leak) introduced in `1.100`. Will fix today and let you know.

**Comment by vsespb** [comment]
fixed in `1.102` (which is in `master`). closing the ticket. let me know if problem re-appear - I will reopen.
and thank you for bug report!

**Comment by kevinjqiu** [comment]
That was fast!  Thanks!  I'll test it when I get home.
